osquery/tables/system/windows/windows_search.cpp

```diff
@@ -0,0 +1,254 @@
+        case VT_UI8: wss << variant.uhVal.HighPart << variant.uhVal.LowPart; break;
+        case VT_I8: wss << variant.hVal.HighPart << variant.hVal.LowPart; break;
+        case VT_DATE: writeDate(variant.date, wss); break;
+        default:
+            wss << "unhandled variant type " << variant.vt;
+            break;
+        }
+    }
+}
+
+std::string ccomandColumnStringValue(CCommand<CDynamicAccessor, CRowset>& cCommand, DBORDINAL columnIndex) {
+    DBTYPE type;
+    cCommand.GetColumnType(columnIndex, &type);
+
+    std::wstringstream wss;
+
+    DBSTATUS status;
+    cCommand.GetStatus(columnIndex, &status);
+    if (status == DBSTATUS_S_ISNULL)
+    {
+        wss << "NULL";
+    }
+    else if (status == DBSTATUS_S_OK || status == DBSTATUS_S_TRUNCATED)
+    {
+        DBTYPE type;
+        cCommand.GetColumnType(columnIndex, &type);
+        switch (type)
+        {
+        case DBTYPE_VARIANT:
+            writePropVariant(*(static_cast<PROPVARIANT*>(cCommand.GetValue(columnIndex))), wss);
+            break;
+        case DBTYPE_WSTR:
+        {
+            DBLENGTH cbLen;
+            cCommand.GetLength(columnIndex, &cbLen);
+            WCHAR szBuffer[2048];
+            StringCchCopyN(szBuffer, ARRAYSIZE(szBuffer), static_cast<WCHAR*>(cCommand.GetValue(columnIndex)), cbLen / sizeof(WCHAR));
+            wss << szBuffer;
+        }
+        break;
+        case DBTYPE_I1:
+            wss << *static_cast<UCHAR*>(cCommand.GetValue(columnIndex));
+            break;
+        case DBTYPE_UI2:
+            wss << *static_cast<USHORT*>(cCommand.GetValue(columnIndex));
+            break;
+        case DBTYPE_I2:
+            wss << *static_cast<SHORT*>(cCommand.GetValue(columnIndex));
+            break;
+        case DBTYPE_UI4:
+            wss << *static_cast<DWORD*>(cCommand.GetValue(columnIndex));
+            break;
+        case DBTYPE_I4:
+            wss << *static_cast<INT*>(cCommand.GetValue(columnIndex));
+            break;
+        case DBTYPE_UI8:
+            wss << *static_cast<ULONGLONG*>(cCommand.GetValue(columnIndex));
+            break;
+        case DBTYPE_I8:
+            wss << *static_cast<LONGLONG*>(cCommand.GetValue(columnIndex));
+            break;
+        case DBTYPE_DATE:
+            writeDate(*static_cast<DATE*>(cCommand.GetValue(columnIndex)), wss);
+            break;
+        default:
+            wss << "unhandled database type " << type;
+            break;
+        }
+    }
+    else
+    {
+        wss << "error reading column";
+    }
+
+    std::wstring wideStr = wss.str();
+    std::wstring_convert<std::codecvt_utf8<wchar_t>> converter;
+    return converter.to_bytes(wideStr);
+}
+
+std::vector<std::map<std::string, std::string>> executeWindowsSearchQuery(CSession& cSession, const std::string& query) {
+    HRESULT hr;
+    std::vector<std::map<std::string, std::string>> results;
+
+    CCommand<CDynamicAccessor, CRowset> cCommand;
+    hr = cCommand.Open(cSession, query.c_str());
```

Comment:
can anyone confirm that this is read only? could someone perform any kind of write operation with a query?

Reply:
I missed this question. Yes, exactly. The query should be read-only